configure.ml

```diff
@@ -717,20 +723,14 @@ let get_lablgtkdir () =
 
 let check_lablgtk_version () =
   let v, _ = tryrun camlexec.find ["query"; "-format"; "%v"; "lablgtk3"] in
-  (true, v)
-
-(* ejgallego: we wait to do version checks until an official release is out *)
-(*  try
-    let vi = numeric_prefix_list v in
-    (* Temporary hack *)
-    if vi = ["3";"0";"beta3"] then (false, v) else
-    let vi = List.map s2i vi in
-    if vi < [3; 0; 0] then
+  try
+    let vl = numeric_prefix_list v in
+    let vn = List.map int_of_string vl in
+    if vn < [3; 1; 0] then
```

Comment:
This must be `>=`

Reply:
Sorry, the problem is not the comparison, but the beta version package by Debian I'm afraid.

Reply:
@treinen apparently the META from from sid of lablgtk3 does not contain the `version = ..` line.

Reply:
Yes, one has to be careful to use the right upstream `tar.bz2` that contains the correct meta, or indeed tell dune to generate a meta with a git commit hash + version.

Reply:
I removed the "resolved" status so that @treinen will be able to spot this discussion more easily.

Reply:
@gares what precise version are you talking about ? 3.0~beta3-1 from stable ?

I do not tamper with the META files, these are the ones installed by the lablgtk3 upstream. The current one (version 3.1.1-1) looks like this:
''''
$ cat /usr/lib/ocaml/lablgtk3/META 
description = ""
requires = "cairo2 threads"
archive(byte) = "lablgtk3.cma"
archive(native) = "lablgtk3.cmxa"
plugin(byte) = "lablgtk3.cma"
plugin(native) = "lablgtk3.cmxs"
''''

The version number of the **debian** package is slightly changed (note the ~ symbol) but this should not concern the configure script of coq, I hope.

Reply:
There is at least two bugs on the packaging scripts for lablgtk3 in debian @treinen , first, as pointed out by Théo you are not picking the right upstream source, thus your files are incorrect; second, you are not properly invoking `dune build` in `rules`.

Reply:
@ejgallo: There is no override of any build targets in debian rules, so dh_auto_build invokes make. Can't I trust that the Makefile provided by uspstream does the right thing ?

I cannot find a post by Théo about using a different usptream source. Currently I am pulling directly from git, using release tags (see the debian/watch file)

Reply:
ok now I think I understand what you are trying to say, @ejgallego and @gares. I understood @gares saying that one has to be careful about pulling the right version. I understand now that one should pull a tar.gz and not directly from a git release tag, is that right ?

In what concerns the build script, you may look a buildlog to understand what is happening :+1: 
 https://buildd.debian.org/status/fetch.php?pkg=lablgtk3&arch=amd64&ver=3.1.1-1%2Bb1&stamp=1597481130&raw=0 for instance

Reply:
@treinen I see an override of build indeed; tho we can maybe see this briefly in person some of these days ?

Reply:
Actulally you are right about old `makefile` in lablgtk being not good for packagers, now the default make target just produces this:
```
Welcome to lablgtk Dune-based build system. Targets are

  - release:   build lablgtk package in release mode
  - dev:       build in developer mode [requires extra dependencies]
  - dev-all:   build in developer mode [requires extra dependencies]
  - nopromote: dev build but without re-running camlp5 generation
  - opam:      internal, used in CI testing
  - clean:     clean build tree

WARNING: Packagers should not use this makefile, but call dune
directly with the right options for their distribution, see README

```

Reply:
Thanks for  the bug report against the debian package of lablgtk3, @ejgallego, this is fixed now for the latest version in sid.

Concerning the override of the build target, I was looking at 3.1.0-3, and was missing the fact that @glondu had in the meantime uploaded a new version. My bad.

Reply:
Thanks @treinen , I will follow up in Debian BTS; I do believe having a `dh_ocaml_dune` helper would be great, Dune developers would happily add some features to `dune install` for example would that help debian.

We could even instruct Dune to produce Debian packages, but that would be a bit inflexible w.r.t. Debian policy IMO, at least until Dune itself is a bit more generic [then dh_dune could just ship a plugin and everything would be taken care of] Note that dune already knows install files etc...

Reply:
@ejgallego See also https://salsa.debian.org/ocaml-team/dh-ocaml/-/merge_requests/1